lib/rom/sql/relation/reading.rb

```diff
@@ -1025,6 +1025,31 @@ def nullify
           new(dataset.__send__(__method__))
         end
 
+        # Wrap other relations using association names
+        #
+        # @example
+        #   tasks.wrap(:owner)
+        #
+        # @param [Array<Symbol>] names A list with association identifiers
+        #
+        # @return [Wrap]
+        #
+        # @api public
+        def wrap(*names)
+          others = names.map do |name|
+            association = associations[name]
+            opts = association.options
+            definition = association.definition
+            target = association.target
+            new_target = association.view ? target.send(association.view) : target
+            to_wrap = association.class.allocate
+            to_wrap.send(:initialize, definition, opts.merge(target: new_target))
+            to_wrap.wrap
+          end
```

Comment:
This indicates that there's a missing abstraction. It seems like having a new method on association objects that will handle all of this (and provide a meaningful name at the same time) would be a good idea. Could you come up with something?